katsdpcontroller/product_controller.py

```diff
@@ -1141,10 +1151,13 @@ def cleanup(task):
                 await self._launch_telstate()
                 # launch containers for those nodes that require them
                 await self.sched.launch(self.physical_graph, self.resolver)
-                alive = self.check_nodes()
+                alive, died = self.check_nodes()
                 # is everything we asked for alive
                 if not alive:
-                    ret_msg = ("Some nodes in the graph failed to start. "
+                    fail_list = ', '.join(node.logical_node.name for node in died)
```

Comment:
Why not `fail_list = ', '.join(node.logical_node.name for node in died) or 'Some nodes'` ?

Reply:
Ok, done.